spring-vault-core/src/main/java/org/springframework/vault/authentication/VaultTokenLookupException.java

```diff
@@ -15,22 +15,23 @@
  */
 package org.springframework.vault.authentication;
 
-import org.springframework.vault.VaultException;
+import org.springframework.http.HttpStatus;
+import org.springframework.vault.exceptions.VaultHttpException;
 
 /**
  * Exception thrown if a token self-lookup fails via {@code auth/token/lookup-self}.
  *
  * @author Mark Paluch
  * @since 2.0
  */
-public class VaultTokenLookupException extends VaultException {
+public class VaultTokenLookupException extends VaultHttpException {
```

Comment:
We should keep extend from `VaultException`. Changing to `VaultHttpException` makes the assumption that lookup is always related to a `HttpStatus` which doesn't apply in every case.

Reply:
In what case is it not? Sorry just not following.

Reply:
In the case of I/O exceptions (such as read timeout, unresolvable host, too many redirects, SSL errors).